### ISSUE TITLE
Is it still active?

### ISSUE THREAD

**Issue by nevack** [soduto/Soduto#2]
Hi, your project is a great implementation of KDE-Connect protocol, i've been using it daily since summer. 

But it doesn't seem to be maintained. Is there any chance to get some new features?

I'm not a swift dev, so can't help you with implementing them.

Thanks!

**Comment by rarecaller** [comment]
It seems to be the only open-source alternative to apple's "Continuity" when you need to bridge Mac and Android, so I really hope it's still alive. 

By the way, is it compatible with Mojave?

**Comment by nevack** [comment]
> By the way, is it compatible with Mojave?
I use Soduto with Mojave, no issues with it

**Comment by rarecaller** [comment]
Oh, thank you for snap response! Hope this is the real reason for the long period since the last update – there's just no need for it. And would be great to keep using Soduto even in the future on macOS 10.15.

**Comment by nevack** [comment]
> And would be great to keep using Soduto

I've been using KDE-connect with linux for years. Last summer I got a macOS machine and found Soduto. It would be great to keep this project alive.

I don't promise you anything now, but if I'd find enough motivation to learn Swift (I'm not into macOS or iOS development, but Android), I'll fork this project and add some features, implemented in official server (presentation clicker at least, notifications and remote at most).

P.S. KDE already have their Qt desktop client ported and working with Windows (Good!) and Mac (Awful!). There's chance they'll sort out bugs and problems later, but I won't hope for it happening soon.

**Comment by giedrius-stanevicius** [comment]
@nevack It is not necessary to make a fork - I would gladly accept contributions to Soduto repository. Alos feel free to ask any questions regarding Soduto development and its codebse ;)

**Comment by rarecaller** [comment]
> @nevack It is not necessary to make a fork - I would gladly accept contributions to Soduto repository. Also feel free to ask any questions regarding Soduto development and its codebse ;)

You're right, I'd also like to see the continuing development of the **current** project. This is an amazing open-sourced rival to proprietary solutions, but until more people discover Soduto, the fork can only hurt its popularity growth. 

For the same reason, it would be nice to add a reference about Soduto to the KDE Connect [wiki page](https://community.kde.org/KDEConnect). (don't know how to do it myself)

**Comment by nevack** [comment]
> Also feel free to ask any questions regarding Soduto development and its codebse ;) 

Yeah, alright. I thought you're inactive.

> it would be nice to add a reference about Soduto to the KDE Connect

Maybe it would be easier make PR to [KDE/kdeconnect-kde](https://github.com/KDE/kdeconnect-kde) to mention Soduto in README.md. It's also a top search result for `kde connect`

**Comment by yurikoles** [comment]
Also coming to Windows 10: https://docs.microsoft.com/en-us/windows-insider/at-home/whats-new-apps-19h1#your-phone

**Comment by nevack** [comment]
> Also coming to Windows 10: [docs.microsoft.com/en-us/windows-insider/at-home/whats-new-apps-19h1#your-phone](https://docs.microsoft.com/en-us/windows-insider/at-home/whats-new-apps-19h1#your-phone)

It has a custom implementation and doesn't use KDE-Connect protocol

**Comment by seltzered** [comment]
> > Also feel free to ask any questions regarding Soduto development and its codebse ;)
> 
> Yeah, alright. I thought you're inactive.
> 
> > it would be nice to add a reference about Soduto to the KDE Connect
> 
> Maybe it would be easier make PR to [KDE/kdeconnect-kde](https://github.com/KDE/kdeconnect-kde) to mention Soduto in README.md. It's also a top search result for `kde connect`
> 
> https://community.kde.org/Infrastructure/Github_Mirror

No idea if KDE would allow it, but perhaps it could be mentioned in https://community.kde.org/KDEConnect#MacOS 

(I discovered KDE connect and Soduto from looking at a bunch of solutions on alternativeto.net today - thanks for making this! Suprised it all just worked!)

**Comment by orschiro** [comment]
Soduto is great. Please keep maintaining!

**Comment by orschiro** [comment]
By the way, can Soduto support notification mirroring?

**Comment by Sd023** [comment]
> Hi, your project is a great implementation of KDE-Connect protocol, i've been using it daily since summer.
> 
> But it doesn't seem to be maintained. Is there any chance to get some new features?
> 
> I'm not a swift dev, so can't help you with implementing them.
> 
> Thanks!

Works Fine with my new M1 MacBook Pro (Big Sur 11.2.2). No major issues yet

**Comment by Sd023** [comment]
Soduto works fine with my new M1 MacBook Pro (Big Sur 11.2.2), No major issues yet.

**Comment by DeveloperEN** [comment]
> Soduto works fine with my new M1 MacBook Pro (Big Sur 11.2.2), No major issues yet.

Good to hear! I couldn't decide whether to build KDE Connect for macOS or just use Soduto but it seems like I found my solution. Also, I don't want to promise but I am planning to contribute to this project after this semester ends.

**Comment by nevack** [comment]
@DeveloperEN JFYI, I managed to build project on recent macOS after upgrading few native libs.
If you have any progress or ideas, please, share with us!

**Comment by yurikoles** [comment]
Meanwhile, KDE Connect has native macOS nightly builds. [Download page](https://kdeconnect.kde.org/download.html) reference [KDE Binary Factory (Jenkins) page](https://binary-factory.kde.org/view/MacOS/job/kdeconnect-kde_Nightly_macos/).

Works fine for me so far.

<img width="958" alt="Знімок екрана 2021-12-20 о 11 51 14" src="https://user-images.githubusercontent.com/922873/146747732-7ad01c4c-2221-4295-a801-9ba5e2c9bef2.png">

**Comment by avichou** [comment]
> Meanwhile, KDE Connect has native macOS nightly builds. [Download page](https://kdeconnect.kde.org/download.html) reference [KDE Binary Factory (Jenkins) page](https://binary-factory.kde.org/view/MacOS/job/kdeconnect-kde_Nightly_macos/).
> 
> Works fine for me so far.
> 
> <img alt="Знімок екрана 2021-12-20 о 11 51 14" width="958" src="https://user-images.githubusercontent.com/922873/146747732-7ad01c4c-2221-4295-a801-9ba5e2c9bef2.png">

whoa this build looks insane ,im assuming its arm native ? btw why arent we getting any in-app updates,it makes it look as though this project and app arent maintained which doesnt seem to be the case!

**Comment by avichou** [comment]
> Meanwhile, KDE Connect has native macOS nightly builds. [Download page](https://kdeconnect.kde.org/download.html) reference [KDE Binary Factory (Jenkins) page](https://binary-factory.kde.org/view/MacOS/job/kdeconnect-kde_Nightly_macos/).
> 
> Works fine for me so far.
> 
> <img alt="Знімок екрана 2021-12-20 о 11 51 14" width="958" src="https://user-images.githubusercontent.com/922873/146747732-7ad01c4c-2221-4295-a801-9ba5e2c9bef2.png">

file sharing is SO much slower than soduto however,like sharing 10 pics takes minutes on kde vs 5seconds on soduto...

**Comment by Jackwmtr** [comment]
Hi! I'm new to Soduto and mac os in general. Trying to get some sort of continuity, I came across Soduto and installed it. It works very nice, much better than KDE connect app on mac OS side, but looks like I can't get an SMS notification for some reason. Call notify is ok. Could someone point me to the right directions, or is it an issue so far?

**Comment by Jackwmtr** [comment]
> > Meanwhile, KDE Connect has native macOS nightly builds. [Download page](https://kdeconnect.kde.org/download.html) reference [KDE Binary Factory (Jenkins) page](https://binary-factory.kde.org/view/MacOS/job/kdeconnect-kde_Nightly_macos/).
> > Works fine for me so far.
> > <img alt="Знімок екрана 2021-12-20 о 11 51 14" width="958" src="https://user-images.githubusercontent.com/922873/146747732-7ad01c4c-2221-4295-a801-9ba5e2c9bef2.png">
> 
> file sharing is SO much slower than soduto however,like sharing 10 pics takes minutes on kde vs 5seconds on soduto...
Are you managed to get stable connection with KDE connect? I'm suffering from random disconnects on mac os and the only way to get it back is to restart the app.

**Comment by avichou** [comment]
> > And would be great to keep using Soduto
> 
> I've been using KDE-connect with linux for years. Last summer I got a macOS machine and found Soduto. It would be great to keep this project alive.
> 
> I don't promise you anything now, but if I'd find enough motivation to learn Swift (I'm not into macOS or iOS development, but Android), I'll fork this project and add some features, implemented in official server (presentation clicker at least, notifications and remote at most).
> 
> P.S. KDE already have their Qt desktop client ported and working with Windows (Good!) and Mac (Awful!). There's chance they'll sort out bugs and problems later, but I won't hope for it happening soon.

have you learned swift? ahaha

**Comment by avichou** [comment]
> > > Meanwhile, KDE Connect has native macOS nightly builds. [Download page](https://kdeconnect.kde.org/download.html) reference [KDE Binary Factory (Jenkins) page](https://binary-factory.kde.org/view/MacOS/job/kdeconnect-kde_Nightly_macos/).
> > > Works fine for me so far.
> > > <img alt="Знімок екрана 2021-12-20 о 11 51 14" width="958" src="https://user-images.githubusercontent.com/922873/146747732-7ad01c4c-2221-4295-a801-9ba5e2c9bef2.png">
> > 
> > file sharing is SO much slower than soduto however,like sharing 10 pics takes minutes on kde vs 5seconds on soduto...
> Are you managed to get stable connection with KDE connect? I'm suffering from random disconnects on mac os and the only way to get it back is to restart the app. 
> 

nope I got fed up with kde connect and just use soduto ,much better ... although not maintained it would seem

I'm wondering if someone made a fork cuz I'd like to have an apple silicon native build ngl :)

**Comment by Jackwmtr** [comment]
> have you learned swift? ahaha
nope :(
Just a bit of Python.

**Comment by avichou** [comment]
> > have you learned swift? ahaha
> nope :(
> Just a bit of Python.
> 

no worries buddy ;) 

have you found anything besides soduto and kde connect by any chance ?

**Comment by giedrius-stanevicius** [comment]
> I'm wondering if someone made a fork cuz I'd like to have an apple silicon native build ngl :)

What do you expect from native apple silicon support? I doubt it would be any faster as Soduto is not a CPU intensive app.

**Comment by Jackwmtr** [comment]
> just use soduto ,much better ... although not maintained it would seem

How about SMS notifications? Do you receive them via Soduto?

**Comment by Jackwmtr** [comment]
> have you found anything besides soduto and kde connect by any chance ?

  No. Just some silly cloud-based apps. Pushbullet is the best for receiving notifications so far.

**Comment by avichou** [comment]
> > I'm wondering if someone made a fork cuz I'd like to have an apple silicon native build ngl :)
> 
> What do you expect from native apple silicon support? I doubt it would be any faster as Soduto is not a CPU intensive app.

less batttery usage i suppose

**Comment by giedrius-stanevicius** [comment]
> > > I'm wondering if someone made a fork cuz I'd like to have an apple silicon native build ngl :)
> > 
> > 
> > What do you expect from native apple silicon support? I doubt it would be any faster as Soduto is not a CPU intensive app.
> 
> less batttery usage i suppose

Does it use a lot of power to you?

**Comment by avichou** [comment]
> > > > I'm wondering if someone made a fork cuz I'd like to have an apple silicon native build ngl :)
> > > 
> > > 
> > > What do you expect from native apple silicon support? I doubt it would be any faster as Soduto is not a CPU intensive app.
> > 
> > 
> > less batttery usage i suppose
> 
> Does it use a lot of power to you?

actualy i checked several times and it doesnt,its fine.still having it updated to AS would still be apprecited

**Comment by avichou** [comment]
> Meanwhile, KDE Connect has native macOS nightly builds. [Download page](https://kdeconnect.kde.org/download.html) reference [KDE Binary Factory (Jenkins) page](https://binary-factory.kde.org/view/MacOS/job/kdeconnect-kde_Nightly_macos/).
> 
> Works fine for me so far.
> 
> <img alt="Знімок екрана 2021-12-20 о 11 51 14" width="958" src="https://user-images.githubusercontent.com/922873/146747732-7ad01c4c-2221-4295-a801-9ba5e2c9bef2.png">

it's not arm native tho, it runs under rosetta

**Comment by sannidhyaroy** [comment]
> Hi! I'm new to Soduto and mac os in general. Trying to get some sort of continuity, I came across Soduto and installed it. It works very nice, much better than KDE connect app on mac OS side, but looks like I can't get an SMS notification for some reason. Call notify is ok. Could someone point me to the right directions, or is it an issue so far?

@Jackwmtr Notifications will stop working and you probably can blame Apple for it, if you'd like. Since, Soduto isn't in active development, the push notification system is deprecated. I tried to implement the newer `UNUserNotificationCenter` and it works for me so far! You can check out the nightly branch of my Soduto fork if you'd like...

**Comment by risalfajar** [comment]
@sannidhyaroy Last commit is 2 years ago, is the other things that doesn't work anymore?

**Comment by sannidhyaroy** [comment]
> @sannidhyaroy Last commit is 2 years ago, is the other things that doesn't work anymore?

I haven't created a pull request to this repository yet. Since, I personally use Soduto, so I'm trying to fix issues that may arise due to deprecated code. You can get my forked version of Soduto from [here](https://github.com/sannidhyaroy/Soduto/releases/latest). There are also other tools like KDE Connect and NearDrop for macOS, so you have to experiment and pick up whichever feels best to you.

**Comment by risalfajar** [comment]
@sannidhyaroy instead of forking, why don't just create a PR instead?

**Comment by sannidhyaroy** [comment]
> @sannidhyaroy instead of forking, why don't just create a PR instead?

I'm not a Swift developer and this is my first project in macOS app development. The purpose of creating a fork is so that I can attempt to optimize the app, fix bugs and experiment with it. Also since I personally heavily rely on Soduto everyday. The fork isn't for everyone, rather for a minor audience who might feel it works decently on their system.
I don't intend to create a PR soon as my code is pretty messy and I don't want to end up with a bunch of embarrassing failed checks.
